### ISSUE TITLE
json 2.3.1 accepts `{` and `[` as valid JSON

### ISSUE THREAD

**Issue by peterkinmond** [flori/json#453]
`JSON.parse` 2.3.1 accepts `{` and `[` as valid JSON, which caused a `JSON::ParserError` in 2.3.0.

#### 2.3.0

```ruby
> JSON.parse('{') # => JSON::ParserError: 783: unexpected token at '{'
> JSON.parse('[') # => JSON::ParserError: 434: unexpected token at ''
> JSON.parse('') # => JSON::ParserError: 783: unexpected token at ''
> JSON.parse('{}') # => {}
> JSON.parse('[]') # => []
> JSON.parse('{{}') # => JSON::ParserError: 783: unexpected token at '{{}'
```

#### 2.3.1

```ruby
> JSON.parse('{') # => {}
> JSON.parse('[') # => []
> JSON.parse('') # => JSON::ParserError: source is not valid JSON!
> JSON.parse('{}') # => {}
> JSON.parse('[]') # => []
> JSON.parse('{{}') # => JSON::ParserError: unexpected token in object at '{}'!
```

**Comment by marcandre** [comment]
This would indicate that there is something wrong with the test suite, as I added these tests in 5b4f1baef055d2a

**Comment by peterkinmond** [comment]
@marcandre it looks like 5b4f1ba didn't get included in the 2.3.1 release:

2.3.1 lists the sha as 0951d77 which [doesn't include those test files](https://github.com/flori/json/tree/0951d7719c3ee7e35b4e1ae80613ce02fb6f20da/tests/fixtures) (`tests/fixtures/fail29.json`, etc)

**Comment by marcandre** [comment]
How, right, my fix just hasn't been released yet.
I guess that when you were running it in 2.3.0 it was using the C/Java implementation, and in 2.3.1 it is using the Ruby-pure implementation, which had this bug, which was fixed by #425.
In that case, this issue can be fixed as already solved, we'll just hope for a release.

**Comment by peterkinmond** [comment]
Makes sense, thanks! I'll close this issue
